x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts

```diff
@@ -0,0 +1,23 @@
+/*
```

Comment:
We are using a simple middleware to handle side-effects here, as opposed to using `oatSaga`. The idea is that we shouldn't overcomplicate the app from the start. `oatSaga`, though simpler than `redux-saga`, still has a fair amount of complexity in its usage. There is no need to have this complexity now, though over time we may feel like adding some more functionality. The other supporting argument to using this simpler approach is that newcomers to the team will most likely be familiar with Redux and the basic concepts of middlewares, but they won't know about `oatSaga`. Thoughts @oatkiller @kevinlog @paul-tavares @parkiino.

Reply:
![image](https://user-images.githubusercontent.com/35559/73080245-832b7780-3e93-11ea-89da-80c7c90376df.png)
![image](https://user-images.githubusercontent.com/35559/73080269-8de60c80-3e93-11ea-9371-fd1bc201348b.png)
Seems like oatsaga has more adoption

Reply:
@peluja1012 what are some of the main differences here?  It looks like we're still using `appSagaFactory` below which, as far as I can tell uses the existing `createSagaMiddleware`.  Is the plan to drop some of the other functionality from https://github.com/elastic/kibana/blob/master/x-pack/plugins/endpoint/public/applications/endpoint/lib/saga.ts ?

Let me know what I'm missing @oatkiller @paul-tavares @parkiino

Reply:
I don't think we're proposing to remove anything. Just that the alerting feature will (initially anyway) forgo `lib/saga` and instead just make its simple middleware: https://github.com/elastic/kibana/blob/62d158b6894f185145077c61e358f61db4566628/x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts#L16

The logic consists of:
1. call the next middleware
2. Check if the action is a certain action (page loaded)
3. if so, make an http request and dispatch an action when thats done.

Redux middleware, by default, provides the ability to run code on each action. `lib/saga`'s main addition to that is to provide the stream of actions as an async generator.